tests/PathOrderMonotonicTest.cpp

```diff
@@ -0,0 +1,286 @@
+            ++color_id;
+            SVG::Color section_color{ (SVG::Color) (((int) SVG::Color::GRAY) + (color_id % 7)) };
+            for (const auto& path : section)
+            {
+                svgFile.writePolyline(path.vertices, section_color);
+            }
+        }
+        svgFile.writeArrow(Point{ 0, 0 }, monotonic_vec, SVG::Color::BLACK);
+        // Note: SVG writes 'itself' when the object is destroyed.
+    }
+#endif //TEST_PATHS_SVG_OUTPUT
+
+    TEST_P(PathOrderMonotonicTest, SectionsTest)
+    {
+        const auto params = GetParam();
+        const double angle_radians{ std::get<1>(params) };
+        const auto& filename = std::get<0>(params);
+        Polygons polylines;
+        ASSERT_TRUE(getInfillLines(filename, angle_radians, polylines)) << "Input test-file could not be read, check setup.";
+
+        const Point& pt_r = polylines.begin()->at(0);
+        const Point& pt_s = polylines.begin()->at(1);
+        const double angle_from_first_line = std::atan2(pt_s.Y - pt_r.Y, pt_s.X - pt_r.X) + 0.5 * M_PI;
+        const Point monotonic_axis(std::cos(angle_from_first_line) * 1000, std::sin(angle_from_first_line) * 1000);
+        const Point perpendicular_axis{ turn90CCW(monotonic_axis) };
+
+        constexpr coord_t max_adjacent_distance = line_distance + 1;
+        PathOrderMonotonic<ConstPolygonRef> object_under_test(angle_from_first_line, max_adjacent_distance, monotonic_axis * -1000);
+        for (const auto& polyline : polylines)
+        {
+            object_under_test.addPolyline(polyline);
+        }
+        object_under_test.optimize();
+
+        // Collect sections:
+        std::vector<std::vector<PathOrderMonotonic<ConstPolygonRef>::Path>> sections;
+        sections.emplace_back();
+        coord_t last_path_mono_projection = projectPathAlongAxis(object_under_test.paths.front(), monotonic_axis);
+        for (const auto& path : object_under_test.paths)
+        {
+            const coord_t path_mono_projection{ projectPathAlongAxis(path, monotonic_axis) };
+            if (path_mono_projection < last_path_mono_projection && ! sections.back().empty())
+            {
+                sections.emplace_back();
+            }
+            sections.back().push_back(path);
+            last_path_mono_projection = path_mono_projection;
+        }
+
+#ifdef TEST_PATHS_SVG_OUTPUT
+        writeDebugSVG(filename, angle_radians, monotonic_axis, sections);
+#endif //TEST_PATHS_SVG_OUTPUT
+
+        size_t section_a_id = 0;
+        for (const auto& section_a : sections)
+        {
+            ++section_a_id;
+
+            size_t section_b_id = 0;
+            for (const auto& section_b : sections)
+            {
+                ++section_b_id;
+                if (section_a_id >= section_b_id)
+                {
+                    continue; // <-- So section B will always be 'later' than section A.
+                }
+
+                // Already tested for A start < B start in the monotonic direction,
```

Comment:
We should remove this comment (and the next line), as it makes little sense anymore.

Reply:
Indeed. Done here: 57ad843b67160b7055544f98b36cf81416301934